src/prefect/blocks/core.py

```diff
@@ -218,9 +225,9 @@ def block_initialization(self) -> None:
     _block_schema_id: Optional[UUID] = None
     _block_schema_capabilities: Optional[List[str]] = None
     _block_schema_version: Optional[str] = None
-    _block_document_id: Optional[UUID] = None
-    _block_document_name: Optional[str] = None
-    _is_anonymous: Optional[bool] = None
+    _block_document_id: Optional[UUID] = PrivateAttr(default=None)
+    _block_document_name: Optional[str] = PrivateAttr(default=None)
+    _is_anonymous: Optional[bool] = PrivateAttr(default=None)
```

Comment:
I marking these fields as `PrivateAttr` might be causing your test failures. I think it's a good idea to explore using `PrivateAttr`, but I think we can do that in a different PR.

```suggestion
    _block_document_id: Optional[UUID] = None
    _block_document_name: Optional[str] = None
    _is_anonymous: Optional[bool] = None
```

Reply:
yep that was indeed the case - removed for now